sled-agent/src/bootstrap/pre_server.rs

```diff
@@ -69,7 +70,10 @@ impl BootstrapAgentStartup {
         // exist before entering any zones.
         illumos_utils::running_zone::ensure_contract_reaper(&base_log);
 
-        let log = base_log.new(o!("component" => "BootstrapAgentStartup"));
+        let log = base_log.new(o!(
+            "component" => "BootstrapAgentStartup",
+            "stage" => SWITCH_ZONE_INIT_STAGE,
```

Comment:
Same question as the bootstore - doesn't this run on all sleds, not just scrimlets?

Reply:
Agreed that this is similar to the bootstore case. Bootstore + bootstrap agent initialization are both **prerequisites** for the switch zone initializing, but:

- They continue executing **after** the switch zone has initialized
- They perform actions unrelated to switch zone initialization

As a more extreme example, I also wouldn't label these as "prerequisites for CockroachDB" or "necessary to launch your first instance" in the logs -- these would be **technically true**, but only because we're applying a particular lens to the chain of dependencies.

I wonder if it would be more useful to enumerate these dependencies, and have them be called out + summarized in the logs more explicitly -- IMO it might be easier periodically see something in the logs like:

```
Switch Zone Initialization:
 [ COMPLETE ] Bootstrap Agent Initialization
 [ INCOMPLETE ] Bootstore Peers Initialized
...
Switch Zone Initialization:
 [ COMPLETE ] Bootstrap Agent Initialization
 [ COMPLETE ] Bootstore Peers Initialized
 [ INCOMPLETE ] Trust Quorum Established
...
Switch Zone Initialization:
 [ COMPLETE ] Bootstrap Agent Initialization
 [ COMPLETE ] Bootstore Peers Initialized
 [ COMPLETE ] Trust Quorum Established
 [ INCOMPLETE ] Switch Zone Launched
```

This is just an example, the dependencies might take a different shape. But I think this call-out -- which could end once the switch zone launches! -- might be more useful than having this "switch zone initialization" label which lasts forever.